include/El/core/AbstractMatrix/decl.hpp

```diff
@@ -47,6 +47,26 @@ class AbstractMatrix
     /** @brief Destroy the AbstractMatrix. */
     virtual ~AbstractMatrix() = default;
 
+    ///@}
+    /** @name Abstract Copies. */
+    ///@{
+
+    /** @brief Copy the underlying matrix into a new matrix with the
+     *         same type and device.
+     *
+     *  @return A newly-allocated matrix that is a deep copy of this
+     *          matrix with the same device type.
+     *  @warning This could dynamically allocate memory.
+     */
+    virtual std::unique_ptr<AbstractMatrix<T>> DeepCopy() const = 0;
```

Comment:
Would it be better to call this `Copy` for consistency with `AbstractDistMatrix`? It's not perfect though since `AbstractDistMatrix::Copy` returns a raw pointer instead of a `unique_ptr` (although `Construct` has the same issue).

Reply:
Done.